QEfficient/generation/text_generation_inference.py

```diff
@@ -222,3 +264,109 @@ def cloud_ai_100_exec_kv(
         print("E2E:", round(total_perf, 2), "tok/s")
         print("Total (E2E) inference time is=", round(total_time, 2), "s")
     print("=============================================================")
+
+
+def cloud_ai_100_exec_kv_helper_loop(
+    batch_size,
+    tokenizer: Union[PreTrainedTokenizer, PreTrainedTokenizerFast],
+    qpc_path: str,
+    prompt: Optional[List[str]] = None,
+    device_id: List[int] = [0],
+    input_len: Optional[int] = None,
+    generation_len: Optional[int] = None,
+    enable_debug_logs: bool = False,
+    stream: bool = True,
+    write_io_dir: Optional[str] = None,
+    automation=False,
+):
+
+    if batch_size == 1:
+        prefill_time = []
+        decode_perf = []
+        total_perf = []
+        total_time = []
+        generated_texts = []
+        for i in range(len(prompt)):
+            latency_stats = cloud_ai_100_exec_kv_helper(
+                tokenizer=tokenizer,
+                prompt=[prompt[i]],
+                qpc=qpc_path,
+                device_id=device_id,
+                input_len=input_len,
+                generation_len=generation_len,
+                enable_debug_logs=enable_debug_logs,
+                stream=stream,
+                write_io_dir=write_io_dir,
+            )
+            generated_texts.append(latency_stats[0])
+            prefill_time.append(latency_stats[1])
+            decode_perf.append(latency_stats[2])
+            total_perf.append(latency_stats[3])
+            total_time.append(latency_stats[4])
+
+        prefill_time = np.average(prefill_time)
+        decode_perf = np.average(decode_perf)
+        total_perf = np.average(total_perf)
+        total_time = np.average(total_time)
+
+    else:
+        latency_stats = cloud_ai_100_exec_kv_helper(
+            tokenizer=tokenizer,
+            prompt=prompt,
+            qpc=qpc_path,
+            device_id=device_id,
+            input_len=input_len,
+            generation_len=generation_len,
+            enable_debug_logs=enable_debug_logs,
+            stream=stream,
+            write_io_dir=write_io_dir,
+        )
+        generated_texts, prefill_time, decode_perf, total_perf, total_time = latency_stats
+
+    print_latency_stats_kv(
+        prompt,
+        generated_texts,
+        batch_size,
+        prefill_time,
+        decode_perf,
+        total_perf,
+        total_time,
+        automation=automation,
+    )
+
+def cloud_ai_100_exec_kv(
+    tokenizer: Union[PreTrainedTokenizer, PreTrainedTokenizerFast],
+    qpc_path: str,
+    prompt: Optional[str] = None,
+    prompts_txt_file_path: Optional[str] = None,
+    device_id: List[int] = [0],
+    input_len: Optional[int] = None,
+    generation_len: Optional[int] = None,
+    enable_debug_logs: bool = False,
+    stream: bool = True,
+    write_io_dir: Optional[str] = None,
+    automation=False,
+):
+    if prompts_txt_file_path is not None:
```

Comment:
Create a function for line 350 to 358, that takes `prompt`, `prompt_file_path`, `batch_size` as parameters and returns `prompts` as output, errors out if inputs are wrong or `batch_size` is not matching `len_prompts`,
Reuse this for line 84-93 in `infer.py`